### ISSUE TITLE
Problems with networkx and drawing Plasma graphs

### ISSUE THREAD

**Issue by unoebauer** [tardis-sn/tardis#501]
This error report is based on **networkx v1.11**:

When trying to generate a Plasma Graph from the LegacyPlasmaArray using the write_to_tex method, it crashes with the error message:

``` python
/home/ulrich/python-virtualenv/tardis/lib/python2.7/site-packages/tardis_sn-1.5.dev2074-py2.7-linux-x86_64.egg/tardis/plasma/base.pyc in write_to_dot(self, fname, latex_label)
    209                     print_graph.node[str(node)]['label']+='$'
    210 
--> 211         nx.write_dot(print_graph, fname)
    212 
    213     def write_to_tex(self, fname_graph):

AttributeError: 'module' object has no attribute 'write_dot'
```

Looking at the documentation of networkx [v1.10](http://networkx.readthedocs.org/en/networkx-1.10/reference/drawing.html) and [v1.11](http://networkx.readthedocs.org/en/networkx-1.11/reference/drawing.html), write_dot is not a top level method of the networkx module. Instead it resides under

```
networkx.drawing.nx_pydot.write_dot
```

When replacing

```
nx.write_dot(print_graph, fname)
```

with

```
nx.drawing.nx_pydot.write_dot(print_graph, fname)                       
```

on line 211 in `tardis/plasma/base.py` and after installing the `pydotplus` package, it works again. However, the generate tex file has also some syntax errors.

**Bottom line:**
- write_to_dot/write_to_tex is incompatible with recent versions of networkx
- the generated tex file has syntax errors

**Comment by yeganer** [comment]
Does the Tex file have real syntax errors or are you talking about
describtions of the plasma graph being displayed wrong/not in math/etc.?

Ulrich Noebauer <notifications@github.com> schrieb am Do., 3. März 2016 um
14:39 Uhr:

**Comment by unoebauer** [comment]
Things like:

```
! Extra }, or forgotten $.
l.212 ...[right] {$dfrac{pi e^{2}}{m_{e} c}f_{lu}}
```

**Comment by yeganer** [comment]
Have you looked at the code. Can those come from errors while parsing the
latex_name of Plasma modules?

Ulrich Noebauer <notifications@github.com> schrieb am Do., 3. März 2016 um
14:46 Uhr:

**Comment by wkerzendorf** [comment]
@aoifeboyle any suggestion?

**Comment by aoifeboyle** [comment]
Strange! It definitely worked fine before. Maybe something has been accidentally removed, or what I added was not consistent for other networkx versions. Leave it with me, I'll try to replicate the issue and see what I can do.

**Comment by yeganer** [comment]
From the code snippet that @unoebauer posted it seems like some math formulas were not parsed correctly. If I remember correctly I had the same issue when I tried building the graph the last time(the syntax errors) but I think those were only a few and a simple search/replace solved it.

**Comment by aoifeboyle** [comment]
@unoebauer Ok, I had a look at this just now, and I don't get any error messages and the graph outputs and compiles to pdf fine. I'm using networkx version 1.9.1 though. So what's the right thing to do in this situation @wkerzendorf? Insist on one particular version of networkx and stick with it?

**Comment by wkerzendorf** [comment]
we should fix it at some stage for 1.11, but for now force 1.9 in conda.

**Comment by unoebauer** [comment]
Closing due to inactivity
